File: backend/src/main/java/nextpos/app/nextpos/model/entity/BarcodeScanner.java
package nextpos.app.nextpos.model.entity;

import lombok.*;
import nextpos.app.nextpos.model.enums.ScannerStatus;
import nextpos.app.nextpos.model.enums.ScannerType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import java.time.LocalDateTime;

@Entity
@Table(name = "barcode_scanners")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BarcodeScanner {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String scannerId; // Unique identifier for the scanner

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ScannerType type; // MOBILE, USB, BLUETOOTH, WIFI

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ScannerStatus status; // ACTIVE, INACTIVE, OFFLINE

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "warehouse_id")
    private Warehouse warehouse;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User assignedUser;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime lastConnectedAt;

    private String ipAddress; // For WiFi scanners
    private String macAddress; // For Bluetooth scanners

    @Column(name = "company_id", nullable = false)
    private Long companyId;
}

File: frontend/src/types/barcodeScanner.ts
export interface ScannerRegistrationRequest {
  scannerName: string;
  scannerType: "MOBILE" | "USB" | "BLUETOOTH" | "WIFI";
  warehouseId: number; // Changed to match backend
  assignedUserId: number; // Changed to match backend
  companyId: number;
  ipAddress?: string;
  macAddress?: string;
}

export interface ScannerRegistrationResponse {
  scannerId: string;
  status: string;
  message: string;
}

export interface BarcodeScanRequest {
  scannerId: string;
  barcode: string;
  companyId: number;
  // Remove warehouseId and userId since backend gets them from scanner
}

export interface BarcodeScanResponse {
  scannerId: string;
  barcode: string;
  productId: number;
  productName: string;
  productSku: string;
  price: number;
  stockQuantity: number;
  success: boolean;
  errorMessage?: string;
  timestamp: string;
}

export interface BarcodeScanner {
  id: number;
  scannerId: string;
  name: string;
  type: "MOBILE" | "USB" | "BLUETOOTH" | "WIFI";
  status: "ACTIVE" | "INACTIVE" | "OFFLINE";
  warehouseId: number;
  assignedUserId: number;
  companyId: number;
  createdAt: string;
  lastConnectedAt: string;
  ipAddress?: string;
  macAddress?: string;
}

export interface ScannerStatusUpdateRequest {
  scannerId: string;
  status: string;
}

export interface ScannerStatusResponse {
  scannerId: string;
  status: string;
  message: string;
}

export interface ScannerDisconnectRequest {
  scannerId: string;
}

export interface ScannerDisconnectResponse {
  scannerId: string;
  status: string;
  message: string;
}
